src/prpy/serialization.py

```diff
@@ -380,11 +426,14 @@ def deserialize_kinbody_state(body, data):
                 body.GetName(), key, e.message
             )
             raise
-
-    body.SetLinkTransformations(
-        map(deserialize_transform, data['link_transforms']),
-        data['dof_branches']
-    )
+    
+    body.SetDOFValues(data['dof_values'])
+    
+    if 'link_transforms' in data:
+        body.SetLinkTransformations(
+            map(deserialize_transform, data['link_transforms']),
+            data['dof_branches']
+        )
```

Comment:
Please remove `SetDOFValues` and restore `SetLinkTransformations`: `SetDOFValues(GetDOFValues(...))` is not an identity-preserving operation.